src/component/framework/collection.js

```diff
@@ -1,8 +1,18 @@
+import {computedFrom} from 'aurelia-framework';
 export class Collection {
 
   activate(model) {
-    this.element = model.element;
-    this.models  = model.value;
-  }
+    this.models   = model.value;
+    this.element  = model.element;
 
+    /* @note the collection element sets index on every element, consider the
+     * index property on the schema reserved for the collection element */
+    this.schema = (index) => {
+      return this.element.schema.map(element => {
+        let el = Object.create(element);
+        el.index = index;
```

Comment:
This is not allowed by eslint and it will return the index instead of the object reference